File: src/main/java/de/blazemcworld/fireflow/node/impl/player/KillPlayerNode.java
package de.blazemcworld.fireflow.node.impl.player;

import de.blazemcworld.fireflow.node.Node;
import de.blazemcworld.fireflow.node.annotation.FlowSignalInput;
import de.blazemcworld.fireflow.node.annotation.FlowSignalOutput;
import de.blazemcworld.fireflow.node.annotation.FlowValueInput;
import de.blazemcworld.fireflow.value.PlayerValue;
import de.blazemcworld.fireflow.value.SignalValue;
import net.minestom.server.entity.Player;

public class KillPlayerNode extends Node {
    public KillPlayerNode() {
        super("Kill Player");

        input("Signal", SignalValue.INSTANCE);
        input("Player", PlayerValue.INSTANCE);
        output("Next", SignalValue.INSTANCE);

        loadJava(KillPlayerNode.class);
    }

    @FlowSignalInput("Signal")
    private static void run() {
        Player p = player().resolve();
        if (p != null) p.kill();
        next();
    }

    @FlowValueInput("Player")
    private static PlayerValue.Reference player() {
        throw new IllegalStateException();
    }

    @FlowSignalOutput("Next")
    private static void next() {
        throw new IllegalStateException();
    }
}

File: src/main/kotlin/de/blazemcworld/fireflow/node/impl/player/action/KillPlayerNode.kt
package de.blazemcworld.fireflow.node.impl.player.action

import de.blazemcworld.fireflow.node.BaseNode
import de.blazemcworld.fireflow.node.NodeContext
import de.blazemcworld.fireflow.node.PlayerType
import de.blazemcworld.fireflow.node.SignalType
import net.minestom.server.item.Material

object KillPlayerNode : BaseNode("Kill Player", Material.SKELETON_SKULL) {
    private val signal = input("Signal", SignalType)
    private val player = input("Player", PlayerType)
    private val next = output("Next", SignalType)

    override fun setup(ctx: NodeContext) {
        ctx[signal].signalListener = { eval ->
            eval[ctx[player]]?.resolve()?.kill()
            eval.emit(ctx[next])
        }
    }
}